pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -300,6 +315,36 @@ func (f *framework) RunScorePlugins(pc *PluginContext, pod *v1.Pod, nodes []*v1.
 	return pluginToNodeScoreMap, nil
 }
 
+// RunNormalizeScorePlugins runs the normalize score plugins. It should be called after
+// RunScorePlugins with the PluginToNodeScoreMap result. It then modifies the map with
+// normalized scores. It returns a non-success Status if any of the normalize score plugins
+// returns a non-success status.
+func (f *framework) RunNormalizeScorePlugins(pc *PluginContext, pod *v1.Pod, scores PluginToNodeScoreMap) *Status {
```

Comment:
An alternative is to run NormalizeScore plugins in the same `RunScorePlugins`. That would helps save on memory by not passing a large map around, but it would reduce parallelism. I haven't gone through all the issues that it could introduce, so I leave it up to you.

```
```

Reply:
I have discussed this with @ahg-g. As you said, there is a trade-off between saving memory vs. reducing parallelism. I have chosen to optimize for parallelism. Since I am new to scheduler, I don't have enough data points to prove that the tradeoff I made is optimal.

I will keep it as is unless you have further comments.

Reply:
Weight is currently applied in ```RunScorePlugins```. We need to change that and apply the weight after NormalizeScore, if a NormalizeScore plugin exist.

If we are going with this current code structure, we need to add a ApplyScoreWeights function that gets called after RunNormalizeScorePlugins in generic_scheduler.go.

Reply:
> That would helps save on memory by not passing a large map around

This is not a problem since passing a map to another method only copies the pointer to the map which is only 8 bytes.

Reply:
> > That would helps save on memory by not passing a large map around
> 
> This is not a problem since passing a map to another method only copies the pointer to the map which is only 8 bytes.

I think the suggestion was to tradeoff parallelism for space. Basically run the plugins sequentially, parallelising across nodes only. If we do this, then we just need to maintain two N-sized arrays instead kN arrays (where k is the number of plugins)